### PR TITLE
e2e(FR-2031): add comprehensive file upload E2E tests for VFolder

### LINKED ISSUE #5264: Add comprehensive file upload E2E tests for VFolder
Comprehensive E2E test coverage for BAIFileExplorer file upload functionality in VFolders.

## Why This Work is Needed

The file upload functionality in VFolders is a critical user workflow with multiple interaction patterns (button-based, drag-and-drop) and edge cases (duplicates, permissions, subdirectories). Currently lacking comprehensive E2E test coverage for these scenarios, which increases the risk of regressions when modifying the file upload implementation.

## Expected Outcomes

- Complete E2E test coverage for file upload scenarios: button-based upload, drag-and-drop, duplicate handling, permission checks, and subdirectory uploads
- Reusable test utilities (selectPropertyFilter helper) for VFolder filtering
- Test plan documentation for future maintenance and expansion

JIRA Issue: FR-2031

### PR DESCRIPTION
**Pull request by yomybaby**
Resolves #5264 ([FR-2031](https://lablup.atlassian.net/browse/FR-2031))

## Summary

Adds comprehensive E2E test coverage for VFolder file upload functionality:

- **Button-based upload**: Single and multiple file uploads via Upload button
- **Drag-and-drop upload**: File upload via drag-and-drop with overlay
- **Duplicate detection**: Confirmation modal when uploading existing files
- **Permission checks**: Upload UI only available with Read & Write permissions
- **Subdirectory upload**: Folder upload preserving directory structure
- **Test utilities**: Refactored `selectPropertyFilter` helper for reusable VFolder filtering
- **BAIPropertyFilter**: Added `aria-label` attributes for better test accessibility

## Test plan

- [ ] All new E2E tests pass on CI
- [ ] Existing vfolder tests remain passing
- [ ] `selectPropertyFilter` utility works correctly across test suites

[FR-2031]: https://lablup.atlassian.net/browse/FR-2031?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/backend.ai-ui/src/components/BAIPropertyFilter.tsx [hunk 1]
```diff
@@ -383,6 +386,7 @@ const BAIPropertyFilter: React.FC<BAIPropertyFilterProps> = ({
               onSearch={onSearch}
               allowClear
               status={!isValid && isFocused ? 'error' : undefined}
+              aria-label="Filter value search"
             />
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply packages/backend.ai-ui/src/components/BAIPropertyFilter.tsx:390]: This `aria-label` is not user-facing display text — it's an internal attribute on a Select component for E2E test automation selectors. Localizing it via i18n would actually break the test selectors since they need a stable, language-independent string. The aria-label here serves as a test hook rather than an accessibility feature for end users.

File: e2e/utils/test-util.ts [hunk 2]
```diff
@@ -411,38 +444,38 @@ export async function deleteForeverAndVerifyFromTrash(
...
-  await expect(
-    page
-      .getByRole('cell', { name: `VFolder Identicon ${folderName}` })
-      .filter({ hasText: folderName }),
-  ).toHaveCount(0);
+  // Verify the folder is either deleted (not visible) or in DELETE-ONGOING status
+  const folderRowAfterDelete = page.getByRole('row').filter({
+    has: page.getByRole('cell', { name: `VFolder Identicon ${folderName}` }),
+  });
+  await expect(folderRowAfterDelete).toBeHidden();
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/utils/test-util.ts:479]: The suggested fallback pattern violates the project's E2E test guidelines which explicitly forbid visibility checks with fallback logic (see `.claude/agents/playwright-test-healer.md`). The strict `.toBeHidden()` assertion is intentional — if the row is still visible after deletion notification has appeared and disappeared, the test should fail to surface a real bug rather than silently accepting a potentially broken state.

File: e2e/vfolder/file-upload.spec.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      ]);
+
+      // 5. Upload multiple test files
+      await fileChooser.setFiles([testFile2Path, testFile3Path]);
+
+      // 6. Verify all uploaded files appear in the file table
+      await modal.verifyFileVisible(path.basename(testFile2Path));
+      await modal.verifyFileVisible(path.basename(testFile3Path));
+
+      // 7. Close modal
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/vfolder/file-upload.spec.ts:158]: This is intentional. The suite uses `test.describe.serial` which guarantees sequential execution — the multi-file upload test reuses the VFolder created by the first test. Creating a separate VFolder per test would add unnecessary overhead (VFolder creation + cleanup) and slow down the suite. The serial guarantee means if the first test fails, all subsequent tests are automatically skipped.

File: e2e/vfolder/file-upload-duplicate.spec.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    });
+
+    test('User can cancel duplicate file upload', async ({ page }) => {
+      // 1. Open the same VFolder in FolderExplorerModal
+      const modal = await openFolderExplorer(page, testFolderName);
+
+      // 2. Verify the file from previous test exists in the table
+      const fileName = path.basename(testFilePath);
+      await modal.verifyFileVisible(fileName);
+
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/vfolder/file-upload-duplicate.spec.ts:156]: Same reasoning as the file-upload.spec.ts serial suite — this suite uses `test.describe.serial` intentionally. The first test creates the VFolder and uploads the baseline file, and the second test verifies the duplicate detection. Creating the VFolder + initial upload in each test would double the test execution time.

File: e2e/vfolder/file-upload-subdirectory.spec.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      await createFolderDialog.getByRole('button', { name: 'Create' }).click();
+
+      // Verify the subfolder appears in the file table (this implicitly waits for folder creation)
+      await modal.verifyFileVisible(subfolderName);
+
+      // 5. Navigate into the newly created folder (click folder name)
+      await page.getByRole('cell', { name: subfolderName }).first().click();
+
+      // Wait for navigation to complete
+      await page.waitForTimeout(1000);
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/vfolder/file-upload-subdirectory.spec.ts]: Good point. Replaced `waitForTimeout(1000)` with a deterministic assertion that waits for the breadcrumb to show the subfolder name. Fixed in the latest commit.

File: e2e/vfolder/file-upload-subdirectory.spec.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      await modal.verifyFileVisible(fileName);
+
+      // 9. Navigate back to root (click home in breadcrumb)
+      // The breadcrumb has a home icon at the beginning
+      const breadcrumb = page.locator('.ant-breadcrumb');
+      await breadcrumb.locator('a').first().click();
+
+      // Wait for navigation back to root
+      await page.waitForTimeout(1000);
+
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/vfolder/file-upload-subdirectory.spec.ts]: Good point. Removed the `waitForTimeout(1000)` and now rely on `modal.verifyFileVisible(subfolderName)` which already uses assertion-based waiting to confirm we're back at root. Fixed in the latest commit.

File: e2e/utils/classes/vfolder/FolderExplorerModal.ts [hunk 7]
```diff
@@ -99,4 +99,10 @@ export class FolderExplorerModal {
     await expect(closeButton).toBeVisible();
     return closeButton;
   }
+
+  async verifyFileVisible(fileName: string): Promise<void> {
+    await expect(this.modal.getByRole('cell', { name: fileName })).toBeVisible({
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply e2e/utils/classes/vfolder/FolderExplorerModal.ts]: Good catch. Added `exact: true` to prevent partial name matches. Fixed in the latest commit.

File: packages/backend.ai-ui/src/components/BAIPropertyFilter.tsx [hunk 8]
```diff
@@ -342,6 +344,7 @@ const BAIPropertyFilter: React.FC<BAIPropertyFilterProps> = ({
           showSearch={{
             optionFilterProp: 'label',
           }}
+          aria-label="Filter property selector"
         />
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply packages/backend.ai-ui/src/components/BAIPropertyFilter.tsx:348]: Same as above — this `aria-label` is used as a stable test automation selector, not as user-facing assistive text for screen readers. Localizing it would break the E2E test selectors that depend on a fixed string.
